### PR TITLE
Refs #31014 -- Added srid argument to FromWKB/FromWKT() GIS functions.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/contrib/gis/db/models/functions.py [hunk 1]
```diff
@@ -367,15 +367,21 @@ class ForcePolygonCW(GeomOutputGeoFunc):
 
 
 class FromWKB(GeoFunc):
-    output_field = GeometryField(srid=0)
-    arity = 1
+    arity = 2
```
Comment (felixxm) [inline django/contrib/gis/db/models/functions.py:370]: I think we should subclass `GeomOutputGeoFunc`.

File: django/contrib/gis/db/models/functions.py [hunk 2]
```diff
@@ -367,15 +367,21 @@ class ForcePolygonCW(GeomOutputGeoFunc):
...
     geom_param_pos = ()
 
+    def __init__(self, expression, srid=0, **extra):
+        expressions = [
+            expression,
+            self._handle_param(srid, "srid", int),
+        ]
+        if "output_field" not in extra:
+            extra["output_field"] = GeometryField(srid=srid)
+        super().__init__(*expressions, **extra)
```
Comment (felixxm) [inline django/contrib/gis/db/models/functions.py:380]: `srid` is not supported on Oracle. We must ignore it, e.g.
```python
    def as_oracle(self, compiler, connection, **extra_context):
        source_expressions = self.get_source_expressions()
        clone = self.copy()
        clone.set_source_expressions(source_expressions[:1])
        extra_context["output_field"] = GeometryField(srid=0)
        return super(FromWKB, clone).as_sql(compiler, connection, **extra_context)
```

File: docs/ref/contrib/gis/functions.txt [hunk 3]
```diff
@@ -359,24 +359,34 @@ are returned unchanged.
...
-.. class:: FromWKB(expression, **extra)
+.. class:: FromWKB(expression, srid=0, **extra)
 
 *Availability*: MariaDB, `MySQL
 <https://dev.mysql.com/doc/refman/en/gis-wkb-functions.html#function_st-geomfromwkb>`__,
 Oracle, `PostGIS <https://postgis.net/docs/ST_GeomFromWKB.html>`__, SpatiaLite
 
-Creates geometry from `Well-known binary (WKB)`_ representation.
+Creates geometry from `Well-known binary (WKB)`_ representation. The optional
+``srid`` argument allows to specify the SRID of the resulting geometry.
```
Comment (felixxm) [inline docs/ref/contrib/gis/functions.txt:369]:
```suggestion
``srid`` argument allows to specify the SRID of the resulting geometry.
``srid`` is ignored on Oracle.
```

File: docs/ref/contrib/gis/functions.txt [hunk 4]
```diff
@@ -359,24 +359,34 @@ are returned unchanged.
...
-.. class:: FromWKT(expression, **extra)
+.. class:: FromWKT(expression, srid=0, **extra)
 
 *Availability*: MariaDB, `MySQL
 <https://dev.mysql.com/doc/refman/en/gis-wkt-functions.html#function_st-geomfromtext>`__,
 Oracle, `PostGIS <https://postgis.net/docs/ST_GeomFromText.html>`__, SpatiaLite
 
-Creates geometry from `Well-known text (WKT)`_ representation.
+Creates geometry from `Well-known text (WKT)`_ representation. The optional
+``srid`` argument allows to specify the SRID of the resulting geometry.
```
Comment (felixxm) [inline docs/ref/contrib/gis/functions.txt:385]:
```suggestion
``srid`` argument allows to specify the SRID of the resulting geometry.
``srid`` is ignored on Oracle.
```

File: docs/ref/contrib/gis/functions.txt [hunk 5]
```diff
@@ -359,24 +359,34 @@ are returned unchanged.
...
+
+.. versionchanged:: 5.1
+
+    Support for the ``srid`` argument was added.
```
Comment (felixxm) [inline docs/ref/contrib/gis/functions.txt:389]:
```suggestion
    The ``srid`` argument was added.
```

File: docs/ref/contrib/gis/functions.txt [hunk 6]
```diff
@@ -359,24 +359,34 @@ are returned unchanged.
...
 <https://dev.mysql.com/doc/refman/en/gis-wkb-functions.html#function_st-geomfromwkb>`__,
 Oracle, `PostGIS <https://postgis.net/docs/ST_GeomFromWKB.html>`__, SpatiaLite
 
-Creates geometry from `Well-known binary (WKB)`_ representation.
+Creates geometry from `Well-known binary (WKB)`_ representation. The optional
+``srid`` argument allows to specify the SRID of the resulting geometry.
+
+.. versionchanged:: 5.1
+
+    Support for the ``srid`` argument was added.
```
Comment (felixxm) [inline docs/ref/contrib/gis/functions.txt:373]:
```suggestion
    The ``srid`` argument was added.
```

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1: I left comments. Please add also a release note.

File: django/contrib/gis/db/models/functions.py (hunk 1 above)
Comment (claudep) [reply django/contrib/gis/db/models/functions.py:370]: I don't think so, as this class has no `geo_field`. Or did you try something working?

File: django/contrib/gis/db/models/functions.py (hunk 2 above)
Comment (claudep) [reply django/contrib/gis/db/models/functions.py:380]: I'll add that and trust you, as I'm not able to test anything on Oracle.

Comment (ngnpope) [reply django/contrib/gis/db/models/functions.py:380]: If you want to test on Oracle, try [my django-docker-box fork main branch](https://github.com/ngnpope/django-docker-box/commits/main)… Will finish polishing it off eventually, but it already resolves many issues.

Comment (claudep) [reply django/contrib/gis/db/models/functions.py:380]: That's nice to suggest, Nick. However, I'm not only unable to test with Oracle, I'm also unwilling to use Oracle at all :nerd_face: There are far enough people caring for proprietary software, I'm trying to focus on FOSS-only stuff.

File: django/contrib/gis/db/models/functions.py (hunk 1 above)
Comment (felixxm) [reply django/contrib/gis/db/models/functions.py:370]: Right :+1:

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1:

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**